pkg/services/ngalert/state/manager_private_test.go

```diff
@@ -754,9 +754,9 @@ func TestProcessEvalResults_StateTransitions(t *testing.T) {
 	t.Run("no-data", func(t *testing.T) {
 		rules := map[ngmodels.NoDataState]*ngmodels.AlertRule{
 			ngmodels.NoData:   baseRuleWith(ngmodels.RuleMuts.WithNoDataExecAs(ngmodels.NoData)),
-			ngmodels.Alerting: baseRuleWith(ngmodels.RuleMuts.WithNoDataExecAs(ngmodels.Alerting)),
-			ngmodels.OK:       baseRuleWith(ngmodels.RuleMuts.WithNoDataExecAs(ngmodels.OK)),
-			ngmodels.KeepLast: baseRuleWith(ngmodels.RuleMuts.WithNoDataExecAs(ngmodels.KeepLast)),
```

Comment:
this makes tests pass but this is a hack, which just make all test cases treat these annotations as rule-defined. The real tests that assert handling of NoData result, happen to pass because they create matching annotations. 
We should fix all the test cases instead.